### PR TITLE
Feat/staking delegation v2

### PR DESCRIPTION
**Pull request by iulianpascalau**
Implementing staking and delegation v2

### CODE REVIEW

File: cmd/node/config/config.toml [hunk 1]
```diff
@@ -21,17 +21,17 @@
    RelayedTransactionsEnableEpoch = 2
 
    # PenalizedTooMuchGasEnableEpoch represents the epoch when the penalization for using too much gas will be enabled
-   PenalizedTooMuchGasEnableEpoch = 2
+   PenalizedTooMuchGasEnableEpoch = 1
```
Comment (SebastianMarian) [inline cmd/node/config/config.toml:24]: not 2?

File: cmd/node/config/config.toml [hunk 2]
```diff
@@ -21,17 +21,17 @@
...
 
    # SwitchJailWaitingEnableEpoch represents the epoch when the system smart contract processing at end of epoch is enabled
-   SwitchJailWaitingEnableEpoch = 2
+   SwitchJailWaitingEnableEpoch = 1
```
Comment (SebastianMarian) [inline cmd/node/config/config.toml:27]: not 2?

File: cmd/node/config/config.toml [hunk 3]
```diff
@@ -21,17 +21,17 @@
...
 
    # BelowSignedThresholdEnableEpoch represents the epoch when the change for computing rating for validators below signed rating is enabled
-   BelowSignedThresholdEnableEpoch = 2
+   BelowSignedThresholdEnableEpoch = 1
```
Comment (SebastianMarian) [inline cmd/node/config/config.toml:30]: not 2?

File: cmd/node/config/config.toml [hunk 4]
```diff
@@ -21,17 +21,17 @@
...
 
    # SwitchHysteresisForMinNodesEnableEpoch represents the epoch when the system smart contract changes its config to consider
    # also (minimum) hysteresis nodes for the minimum number of nodes
-   SwitchHysteresisForMinNodesEnableEpoch = 2
+   SwitchHysteresisForMinNodesEnableEpoch = 1
```
Comment (SebastianMarian) [inline cmd/node/config/config.toml:34]: not 2?

**Review by sasurobert — APPROVED** [review]

File: cmd/node/config/systemSmartContractsConfig.toml [hunk 5]
```diff
@@ -1,16 +1,16 @@
 [StakingSystemSCConfig]
     GenesisNodePrice = "2500000000000000000000" #2.5K eGLD
     MinStakeValue = "100000000000000000000" #100 eGLD
+    MinUnstakeTokensValue = "10000000000000000000" #10eGLD
     UnBondPeriod = 250
     MinStepValue = "100000000000000000000"
-    AuctionEnableEpoch = 100000
-    StakeEnableEpoch = 2
+    StakeEnableEpoch = 1
```
Comment (SebastianMarian) [inline cmd/node/config/systemSmartContractsConfig.toml:7]: not 2?

File: cmd/node/config/config.toml (hunk 1 above)
Comment (iulianpascalau) [reply cmd/node/config/config.toml:24]: as agreed, 1 should be here

File: cmd/node/config/config.toml (hunk 2 above)
Comment (iulianpascalau) [reply cmd/node/config/config.toml:27]: as agreed, 1 should be here

**Review by SebastianMarian — APPROVED** [review]

**Review by AdoAdoAdo — DISMISSED** [review]

File: cmd/node/config/systemSmartContractsConfig.toml (hunk 5 above)
Comment (iulianpascalau) [reply cmd/node/config/systemSmartContractsConfig.toml]: done

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
Initial tests passed.

**Review by SebastianMarian — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**